yarn-project/prover-client/src/tx-prover/tx-prover.ts

```diff
@@ -1,43 +1,40 @@
 import { BBNativeRollupProver, TestCircuitProver } from '@aztec/bb-prover';
-import { type L2BlockSource, type ProcessedTx } from '@aztec/circuit-types';
 import {
-  type BlockResult,
+  type BlockProver,
   type ProverClient,
   type ProvingJobSource,
-  type ProvingTicket,
   type ServerCircuitProver,
 } from '@aztec/circuit-types/interfaces';
-import { Fr, type GlobalVariables } from '@aztec/circuits.js';
+import { Fr } from '@aztec/circuits.js';
 import { NativeACVMSimulator } from '@aztec/simulator';
 import { type TelemetryClient } from '@aztec/telemetry-client';
-import { type WorldStateSynchronizer } from '@aztec/world-state';
+import { type MerkleTreeOperations } from '@aztec/world-state';
 
 import { type ProverClientConfig } from '../config.js';
 import { ProvingOrchestrator } from '../orchestrator/orchestrator.js';
 import { MemoryProvingQueue } from '../prover-agent/memory-proving-queue.js';
 import { ProverAgent } from '../prover-agent/prover-agent.js';
 
 /**
- * A prover accepting individual transaction requests
+ * A prover factory.
+ * TODO(palla/prover-node): Rename this class
  */
 export class TxProver implements ProverClient {
-  private orchestrator: ProvingOrchestrator;
   private queue: MemoryProvingQueue;
   private running = false;
 
   private constructor(
     private config: ProverClientConfig,
-    private worldStateSynchronizer: WorldStateSynchronizer,
     private telemetry: TelemetryClient,
     private agent?: ProverAgent,
   ) {
+    // TODO(palla/prover-node): Cache the paddingTx here, and not in each proving orchestrator,
+    // so it can be reused across multiple ones and not recomputed every time.
     this.queue = new MemoryProvingQueue(telemetry, config.proverJobTimeoutMs, config.proverJobPollIntervalMs);
-    this.orchestrator = new ProvingOrchestrator(
-      worldStateSynchronizer.getLatest(),
-      this.queue,
-      telemetry,
-      config.proverId,
-    );
+  }
+
+  public createBlockProver(db: MerkleTreeOperations): BlockProver {
+    return new ProvingOrchestrator(db, this.queue, this.telemetry, this.config.proverId);
```

Comment:
Ohhhh, all prover can share a single queue (and implicitly, the same prover agents) since all the data is generated by the orchestrators :exploding_head:. This is awesome!